engine/src/main/java/com/redhat/cloud/notifications/db/repositories/EndpointRepository.java

```diff
@@ -139,10 +143,18 @@ public boolean incrementEndpointServerErrors(UUID endpointId, int maxServerError
          * It may or may not have been disabled already from the frontend or because of a 4xx error.
          */
         if (endpoint.isPresent() && endpoint.get().isEnabled()) {
-            if (endpoint.get().getServerErrors() + currentServerErrors > maxServerErrors) {
+            LocalDateTime currentTime = LocalDateTime.now(ZoneId.of("UTC"));
+            boolean minDelaySinceFirstErrorRespected = false;
+            if (endpoint.get().getServerErrorsSince() != null) {
+                Duration spentDuration = Duration.between(endpoint.get().getServerErrorsSince(), currentTime);
+                minDelaySinceFirstErrorRespected = spentDuration.compareTo(minDelaySinceFirstServerErrorBeforeDisabling) > 0;
+            }
```

Comment:
This whole block could be under the old `if (endpoint.get().getServerErrors() + currentServerErrors > maxServerErrors)` condition as it only needs to be executed when that condition is `true`.